src/main/java/com/photolocos/enterprise/dao/PhotoDAO.java

```diff
@@ -12,21 +11,19 @@
  */
 @Component
 public class PhotoDAO implements IPhotoDAO {
-
     Map<Integer, PhotoDTO> photos = new HashMap<>();
+    private Set<PhotoDTO> matchedPhotos = new HashSet<>();
+    private List<PhotoDTO> photosCollection = new ArrayList<>(photos.values());
```

Comment:
There's a bit of risk in moving these out of the methods (local variables) and promoting them to attributes.  Each of the methods has an if test.  If that if test evaluates to false, or if the collection is empty, it will end up returning what was previously stored in these variables, which may not be correct.

For that reason, I lean toward keeping these as local variables.